internal/proxy/oauthproxy_test.go

```diff
@@ -263,6 +263,129 @@ func TestNewReverseProxyTLSSkipVerify(t *testing.T) {
 	}
 }
 
+func TestNewReverseProxyPreserveHost(t *testing.T) {
```

Comment:
can we make these table drive tests as well?

Reply:
Almost done w/ these and so far confirmed that replacing a bunch of test functions w/ two bigger table-driven test functions hasn't altered code coverage.

Reply:
Done.